### PR TITLE
Add `message` argument to `Objects.requireNonNull()`

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: api/client/src/main/java/org/projectnessie/client/auth/oauth2/RefreshTokensFlow.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ class RefreshTokensFlow extends AbstractFlow {
 
   @Override
   public Tokens fetchNewTokens(@Nullable Tokens currentTokens) {
-    Objects.requireNonNull(currentTokens);
+    Objects.requireNonNull(currentTokens, "currentTokens is null");
     Objects.requireNonNull(currentTokens.getRefreshToken());
```
Comment (adutra) [inline api/client/src/main/java/org/projectnessie/client/auth/oauth2/RefreshTokensFlow.java:34]: Why not this one?

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Utils.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public final class S3Utils {
   private S3Utils() {}
 
   public static Optional<String> extractBucketName(URI uri) {
-    String scheme = requireNonNull(uri).getScheme();
+    String scheme = requireNonNull(uri, "URI argument missing").getScheme();
```
Comment (adutra) [inline catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Utils.java:35]: Couldn't we settle on a standard message format? E.g.. `argument %s is null`

File: api/client/src/main/java/org/projectnessie/client/auth/oauth2/RefreshTokensFlow.java (hunk 1 above)
Comment (snazy) [reply api/client/src/main/java/org/projectnessie/client/auth/oauth2/RefreshTokensFlow.java:34]: Which one?

Comment (snazy) [reply api/client/src/main/java/org/projectnessie/client/auth/oauth2/RefreshTokensFlow.java:34]: Not sure BTW why the arg is `@Nullable` but the impl requires non-null

Comment (adutra) [reply api/client/src/main/java/org/projectnessie/client/auth/oauth2/RefreshTokensFlow.java:34]: > Which one?

```java
Objects.requireNonNull(currentTokens.getRefreshToken(), "currentTokens.getRefreshToken() is null");
```

> Not sure BTW why the arg is @Nullable but the impl requires non-null

The interface is used both for initial token fetches and token refreshes; `currentTokens` is null for initial token fetches and non-null for token refreshes.

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Utils.java (hunk 2 above)
Comment (snazy) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Utils.java:35]: At least we have something that helps more than just an NPE w/o a reason

**Review by adutra — APPROVED** [review]

**Pull request merged**